src/unix.rs

```diff
@@ -81,61 +88,66 @@ impl Client {
         Ok(Client::from_fds(pipes[0], pipes[1]))
     }
 
-    pub unsafe fn open(s: &str) -> Option<Client> {
-        Client::from_fifo(s).or_else(|| Client::from_pipe(s))
+    pub unsafe fn open(s: &str) -> Result<Client, ErrFromEnv> {
+        match (Self::from_fifo(s), Self::from_pipe(s)) {
```

Comment:
This is a bit unfortunate. Previously if people is using fifo-style jobserver, they don't need to read from pipe. Now they need both.

Reply:
I'll check, if both `fifo:PATH` and `R,W` can be provided, but even if they can and will. Both clients will be created, but only one `from_fifo` will be used. Is smth bad in calling `from_pipe` after succeeded `from_fifo`?

Reply:
>  Is smth bad in calling from_pipe after succeeded from_fifo?

Not really. Just an idea that if we don't need it we don't create it. It also reduces syscalls. Anyway, I am not the maintainer of this project. Just someone cares about it :)